src/Command/NewYamlHelpCommand.cs

```diff
@@ -0,0 +1,252 @@
+﻿using System.Net;
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Globalization;
+using System.IO;
+using System.Management.Automation;
+using System.Management.Automation.Runspaces;
+using System.Text;
+
+using Microsoft.PowerShell.PlatyPS.YamlWriter;
+using Microsoft.PowerShell.PlatyPS.Model;
+
+namespace Microsoft.PowerShell.PlatyPS
+{
+    /// <summary>
+    /// Cmdlet to generate the markdown help for commands, all commands in a module or from a MAML file.
+    /// </summary>
+    [Cmdlet(VerbsCommon.New, "YamlHelp", HelpUri = "https://go.microsoft.com/fwlink/?LinkID=2096483")]
+    [OutputType(typeof(FileInfo[]))]
+    public sealed class NewYamlHelpCommand : PSCmdlet
+    {
+
+        #region parameters
+        [Parameter(Mandatory=true, Position=1, ValueFromPipeline=true, ValueFromPipelineByPropertyName=true)]
+        public string[]? Path { get; set; }
+
+        [Parameter(Mandatory=true)]
+        public string? OutputFolder { get; set; }
+
+        [Parameter]
+        public Encoding Encoding { get; set;} = Encoding.UTF8;
+
+        [Parameter]
+        SwitchParameter Force { get; set; }
+        #endregion
+
+        private string fullPath = string.Empty;
+        private List<CommandHelp> CommandHelpCollection = new List<CommandHelp>();
+        // private TransformSettings transformSettings = null;
+
+    protected override void BeginProcessing()
+    {
+            
+        ProviderInfo provider;
+        var resolvedOutputFolder = this.SessionState.Path.GetResolvedProviderPathFromPSPath(OutputFolder, out provider);
+        if (resolvedOutputFolder is null)
+        {
+            if (WildcardPattern.ContainsWildcardCharacters(OutputFolder))
+            {
+                throw new ArgumentException("OutputFolder must be a single path");
+            }
+            else
+            {
+                Directory.CreateDirectory(OutputFolder);
+            }
+        }
+        else
+        {
+            if (provider.Name != "FileSystem")
+            {
+                throw new ArgumentException("OutputFolder must be a directory");
+            }
+            else if (resolvedOutputFolder.Count > 1)
+            {
+                throw new ArgumentException("OutputFolder must be a single path");
+            }
+            else if (File.Exists(resolvedOutputFolder[0]))
+            {
+                throw new ArgumentException("OutputFolder must be a directory");
+            }
+            else
+            {
+                fullPath = resolvedOutputFolder[0];
+            }
+            Directory.CreateDirectory(fullPath);
+        }
+
+        /*
```

Comment:
Does this need to be here? or should be uncommented out?

Reply:
I wanted to hang on to it, because I thought I might need it later. I can remove it